inc/saigroup.h

```diff
@@ -0,0 +1,826 @@
+/**
+ * Copyright (c) 2014 Microsoft Open Technologies, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License"); you may
+ *    not use this file except in compliance with the License. You may obtain
+ *    a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR
+ *    CONDITIONS OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT
+ *    LIMITATION ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS
+ *    FOR A PARTICULAR PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
+ *
+ *    See the Apache Version 2.0 License for specific language governing
+ *    permissions and limitations under the License.
+ *
+ *    Microsoft would like to thank the following companies for their review and
+ *    assistance with these files: Intel Corporation, Mellanox Technologies Ltd,
+ *    Dell Products, L.P., Facebook, Inc
+ *
+ * @file    saigroup.h
+ *
+ * @brief   This module defines SAI Next Hop, RFP, L2MC, IPMC Group interface
+ */
+
+#if !defined (__SAIGROUP_H_)
+#define __SAIGROUP_H_
+
+#include <saitypes.h>
+
+/**
+ * @defgroup SAIGROUP SAI - Next hop, RPF, L2MC, IPMC group specific API definitions
+ *
+ * @{
+ */
+
+/**
+ * @brief Next hop group type
+ */
+typedef enum _sai_next_hop_group_type_t
+{
+    /** Next hop group is ECMP */
+    SAI_NEXT_HOP_GROUP_TYPE_ECMP,
+
+    /* Other types of next hop group to be defined in the future, e.g., WCMP */
```

Comment:
would there be more types as you have made group as generic object for any collection of egresses.